src/ethereum/ethEngine.js

```diff
@@ -722,6 +723,12 @@ export class EthereumEngine extends CurrencyEngine {
             params[0],
             'latest'
           ])
+          // Convert hex
+          if (!isHex(result.result)) {
+            throw new Error('Infura eth_getBalance not hex')
```

Comment:
Since `bns.add` apparently works fine with both decimal and hex strings, is the check really needed?

Reply:
My concern is if we don’t get a hex string, then the API is broken or we have a malicious or broken server. It’s not much of a check but at least we won’t barf on a value like “123.456” which is a valid bns value but not valid hex.